addons/sale/models/sale_order.py

```diff
@@ -732,6 +733,16 @@ def action_quotation_send(self):
             'context': ctx,
         }
 
+    def _validate_distribution(self):
+        self.ensure_one()
```

Comment:
Why?
Just loop on `self` if it is really needed, but I don't see why by looking at the code.
This function could (should) even be declared on the lines instead, since you validate the distribution on the lines, not the one of the SO.
Then you can keep the same name (`_validate_analytic_distribution`) and call `super` with the right args instead of this `_validate_distribution`, which is a bit too generic.